specifyweb/frontend/js_src/lib/components/QueryComboBox/index.tsx

```diff
@@ -247,6 +252,26 @@ export function QueryComboBox({
     (typeof typeSearch === 'object' ? typeSearch?.table : undefined) ??
     field.relatedTable;
 
+  const [treeDefinition] = useAsyncState(
+    React.useCallback(
+      async () =>
+        resource?.specifyTable === tables.Determination &&
+        resource.collection?.related?.specifyTable === tables.CollectionObject
+          ? (resource.collection?.related as SpecifyResource<CollectionObject>)
+              .rgetPromise('collectionObjectType')
+              .then(
+                (
+                  collectionObjectType:
+                    | SpecifyResource<CollectionObjectType>
+                    | undefined
+                ) => collectionObjectType?.get('taxonTreeDef')
+              )
+          : undefined,
+      [resource, resource?.collection?.related?.get('collectionObjectType')]
+    ),
+    false
+  );
+
   // FEATURE: use main table field if type search is not defined
   const fetchSource = React.useCallback(
     async (value: string): Promise<RA<AutoCompleteItem<string>>> =>
```

Comment:
do you need to add a check in this function for `treeDefinition` being loaded?
i.e don't start fetching resource until treeDefinition is resolved

that also means you need to turn treeDefinition into a tri-state value (some value to indicate that you are still fetching, some value to indicate there is nothing to fetch, and the actual result object)